internal/data/account.go

```diff
@@ -26,13 +27,13 @@ type Account struct {
 // the given slot or nil if none exists.
 func (a *Account) FindCharacterInSlot(slot int) (*Character, error) {
 	var character Character
-	q := db.Model(&a).Related(&character).Where("slot = ?", slot)
+	err := db.First(&character).Where("slot = ?", slot).Where("account_id = ?", &a.ID).Error
```

Comment:
From the docs it looks like the way to add multiple conditions is to use something like this, annoyingly
```suggestion
	err := db.First(&character).Where("slot = ?" AND account_id = ?", slot, a.ID).Error
```

Reply:
It was actually an order issue on top of that. .First() should be called after the .Where()